### DISCUSSION TITLE
Problem with Route Handler in Next 13

### DISCUSSION THREAD

**Discussion by Ramxnchv** [supabase/supabase#13914] [Questions]
Hello, I'm trying to use the createRouteHandlerSupabaseClient in route.ts and it works for making queries like supabase.from(......) but when i try to call supabase.auth.getSession() i receive {data: {session: null}, error: null} which is weird because it is working in server side components and it is working in middleware, so I'm a bit confused about why is this happening. It's like it is losing the auth token cookie from middleware to route handler.

![image](https://user-images.githubusercontent.com/38292919/233844633-42f75ba5-d2f1-4e39-a832-d2b910583c2c.png)
![image](https://user-images.githubusercontent.com/38292919/233844664-5f99caa5-482b-4336-81e5-05aacdfc47df.png)
![image](https://user-images.githubusercontent.com/38292919/233844704-a062eb1f-c1c7-4b17-b6ca-53427240d88f.png)

**Comment by oduncan** [comment]
I'm having a similar issue, the cookie is set on the request and getSession() gives me an empty session, can't be working as intended.

**Comment by StephenTangCook** [comment]
Experiencing the same issue. I posted about it in the discord here: https://discord.com/channels/839993398554656828/1099899130500882512/1099899130500882512
But no direct solution yet.

**Comment by meerkat-citronella** [comment]
Bump. I am also having this issue.

**Comment by michedapt** [comment]
Also stuck on this.

**Comment by kimhwanhoon** [comment]
Me too! why does it work on server component but not on route handler? maybe I have lack of knowlegde how everything works. Or is it just a bug?
